drivers/sensor/stmemsc/CMakeLists.txt

```diff
@@ -0,0 +1,12 @@
+# ST Microelectronics stmemsc hal i/f
+#
+# Copyright (c) 2021 STMicroelectronics
+#
+# SPDX-License-Identifier: Apache-2.0
+#
+zephyr_library()
+
+zephyr_library_sources_ifdef(CONFIG_I2C        stmemsc_i2c.c)
+zephyr_library_sources_ifdef(CONFIG_SPI        stmemsc_spi.c)
+
+zephyr_include_directories(./)
```

Comment:
Is it possible to not add this?

ie, can we do `#include "../stmemsc/stmemsc.h" in the actual drivers?

Reply:
> Is it possible to not add this?
> 
> ie, can we do `#include "../stmemsc/stmemsc.h" in the actual drivers?

Do you think it is better? Of course I don't see any practical issue in doing what you suggest, but it seems less elegant and I would like to understand if there is a reason to not add the stmemsc directory in the include PATH. It seems to me more convenient than creating a dependence in all the drivers (at the end will be many) that will include stmemsc.h

Reply:
@galak 
I have done the requested modification. Still, before submitting it I would like, if possible, to get a rational why should be better in this way. Is there any other already existing example in the source code? Thanks for any comment on this.

OPinions from the other reviewers?

Reply:
The concern is that when we add `zephyr_include_directories` this gets added globally for all files that get built.  So I feel like the more we can limit those cases the better.

@tejlmand do we have any means to limit the scope of `include_dirs` getting added only for specific source files?

Reply:
@galak I re-pushed following your suggestions

Reply:
thanks

Reply:
> @tejlmand do we have any means to limit the scope of `include_dirs` getting added only for specific source files?

Did you try using zephyr_library_include_directories()?

Reply:
@mbolivar-nordic @galak @tejlmand 
To be honest I do not like the <#include "../stmemsc/stmemsc.h"> approach. I would prefer having a different solutions. Now my question is what is the difference between the two macros (i.e. zephyr_library_include_directories() vs zephyr_include_directories()) and what is anyway the concern behind it?
I have to re-push anyway, so it would be good to find the better way to do this stuff.
Thanks for giving me your valuable opinions.